### PR TITLE
Lookup Join Class from static defined models

### PR DESCRIPTION
**Pull request by rishi-aga**
## Description
Lookup Join Class from static defined models. If not available then assume its defined dynamically and use existing logic.
Remove unused imports from table handlebar template.

## How Has This Been Tested?
Existing Tests Pass.
Integration Test added.

## License
I confirm that this contribution is made under an Apache 2.0 license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java [hunk 1]
```diff
@@ -193,4 +194,23 @@ public String trimColumnReferences(String str) {
...
         return str;
     }
+
+    /**
+     * Get Class Name for provided modelName from static models if available else default value.
+     * @param modelName model name.
+     * @return class name.
+     */
+    public String getJoinClassName(String modelName) {
+        return ElideDynamicEntityCompiler.getStaticModelClassName(modelName, EMPTY_STRING,
```
Comment (aklish) [inline elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java:204]: There is a NO_VERSION constant you can use here. you can use here.

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java [hunk 2]
```diff
@@ -193,4 +194,23 @@ public String trimColumnReferences(String str) {
...
+                        capitalizeFirstLetter(modelName));
+    }
+
+    /**
+     * Get Class Import for provided modelName from static models if available.
+     * @param modelName model name.
+     * @return import statement.
+     */
+    public String getJoinClassImport(String modelName) {
+        return ElideDynamicEntityCompiler.getStaticModelClassImport(modelName, EMPTY_STRING, EMPTY_STRING);
```
Comment (aklish) [inline elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java:214]: same comment.

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/compile/ElideDynamicEntityCompiler.java [hunk 3]
```diff
@@ -201,4 +238,38 @@ public ClassLoader getClassLoader() {
...
+
+    /**
+     * Get Class Name for provided (modelName, modelVersion).
+     * @param modelName model name.
+     * @param modelVersion model version.
+     * @param defaultValue default value.
+     * @return class name for provided (modelName, modelVersion) if available else default value.
+     */
+    public static String getStaticModelClassName(String modelName, String modelVersion, String defaultValue) {
+        ModelMapValue value = STATIC_MODEL_DETAILS.get(new ModelMapKey(modelName, modelVersion));
```
Comment (aklish) [inline elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/compile/ElideDynamicEntityCompiler.java:260]: We should use these lookups to validate the joins as well (so we don't get a cryptic error here if they put down a class that doesn't exist).

File: elide-datastore/elide-datastore-aggregation/src/test/resources/configs/models/tables/SalesView.hjson [hunk 4]
```diff
@@ -174,7 +174,7 @@
       [
         {
           name: region
-          to: RegionDetails
+          to: regionDetails
```
Comment (aklish) [inline elide-datastore/elide-datastore-aggregation/src/test/resources/configs/models/tables/SalesView.hjson:177]: What test is validating this change?

**Review by aklish — CHANGES_REQUESTED** [review]

File: elide-datastore/elide-datastore-aggregation/src/test/resources/configs/models/tables/SalesView.hjson (hunk 4 above)
Comment (rishi-aga) [reply elide-datastore/elide-datastore-aggregation/src/test/resources/configs/models/tables/SalesView.hjson:177]: https://github.com/yahoo/elide/blob/hjson_join_validation/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/AggregationDataStoreIntegrationTest.java#L916
https://github.com/yahoo/elide/blob/hjson_join_validation/elide-datastore/elide-datastore-aggregation/src/test/java/com/yahoo/elide/datastores/aggregation/integration/AggregationDataStoreIntegrationTest.java#L989
`customerRegionRegion` field is pulled from static model now. (https://github.com/yahoo/elide/blob/hjson_join_validation/elide-datastore/elide-datastore-aggregation/src/test/resources/configs/models/tables/SalesView.hjson#L57)

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java (hunk 1 above)
Comment (rishi-aga) [reply elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java:204]: 👍

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java (hunk 2 above)
Comment (rishi-aga) [reply elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/parser/handlebars/HandlebarsHelper.java:214]: 👍

File: elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/compile/ElideDynamicEntityCompiler.java (hunk 3 above)
Comment (rishi-aga) [reply elide-contrib/elide-dynamic-config-helpers/src/main/java/com/yahoo/elide/contrib/dynamicconfighelpers/compile/ElideDynamicEntityCompiler.java:260]: Added validation on Join's `to` value

File: elide-contrib/elide-dynamic-config-helpers/src/test/resources/validator/bad_join_model/models/tables/table1.hjson [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{
+  tables: [{
+      name: PlayerStats
+      sql: PlayerStats
+      schema: gamedb
+      cardinality : large
+      hidden : false
+      readAccess : A user is admin or is a player in the game
```
Comment (aklish) [inline elide-contrib/elide-dynamic-config-helpers/src/test/resources/validator/bad_join_model/models/tables/table1.hjson:8]: For any HJSON file that we expect to fail, let's add comments explaining the line we expect to blow up and why.

**Review by aklish — APPROVED** [review]

**Review by aklish — APPROVED** [review]

**Pull request merged**